### PR TITLE
moving to ansible v2.2 compatibility

### PR DESCRIPTION
**Pull request by leseb**
Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
@ktdreyer @alfredodeza @andrewschoen 
@samdoran do we also need to do the `with_together` loops?

**Comment by andrewschoen** [comment]
test this please

**Comment by andrewschoen** [comment]
Looks good to me, I've changed the CI to use ansible==2.2 for that test I just started up.

**Comment by andrewschoen** [comment]
All the tests seem to be failing at:

```
TASK [ceph-mon : stat for ceph config and keys] ********************************
task path: /home/jenkins-build/build/workspace/ceph-ansible-pull-requests/DIST/ceph_ansible_pr_centos7/ceph-ansible/roles/ceph-mon/tasks/docker/checks.yml:13
fatal: [localhost]: FAILED! => {
    "failed": true, 
    "msg": "'ceph_config_keys' is undefined"
}
```

https://jenkins.ceph.com/job/ceph-ansible-pull-requests/792/DIST=ceph_ansible_pr_centos7/console

File: roles/ceph-mon/tasks/docker/checks.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@
 - name: stat for ceph config and keys
   stat:
     path: "{{ item }}"
-  with_items: ceph_config_keys
+  with_items: "{{ ceph_config_keys }}"
```
Comment (samdoran) [inline roles/ceph-mon/tasks/docker/checks.yml]: You can also use `default([])` inline. This has the effect of [skipping the task](http://docs.ansible.com/ansible/playbooks_conditionals.html#loops-and-conditionals) if the variable is undefined.

```yaml
with_items: " {{ cepth_config_keys | default([]) }}"
```

Comment (samdoran) [inline roles/ceph-mon/tasks/docker/checks.yml]: Very true!

**Comment by leseb** [comment]
@andrewschoen , @ktdreyer @alfredodeza Alright, it's a critical move, once we have this we need to branch out for "1.9.4 to 2.1.x" and explain this won't bring more features into the other branch, just bug fixes.

**Comment by jfcoz** [comment]
Signed-off-by: Julien Francoz <julien@francoz.net>

**Comment by leseb** [comment]
thanks @jfcoz

**Comment by ktdreyer** [comment]
It would be a good idea to create a "version-1" branch at this point to maintain Ansible 1.9 compatibility, and then master can be tagged with "v2.0", "v2.1", etc.

**Comment by leseb** [comment]
@ktdreyer I just pushed a branch called: ansible-1.9, how does that sound for you?
Can we merge this now?

**Comment by ktdreyer** [comment]
@leseb, would you please clarify what tags we should expect to see on each branch, going forward?

For example, will "1.0.z" tags be on the ansible-1.9 branch? If so, what tags will be on master?

**Comment by leseb** [comment]
@ktdreyer well going forward the ansible-1.9 branch will only receive bug fixes so I haven't thought about doing any tags for it. I'd like to keep our focus on 2.x, Ansible 2.x has been around for more than almost a year now, and most of the distro are shipping it so I believe it's fair.

In order to not confuse people and try to stay consistent I think next's master tag should be v2.x. What do you think?

**Comment by andrewschoen** [comment]
I did a quick test and the ``ansible-1.9`` branch isn't compatible with ansible 1.9.4.

I get:

```
ERROR: yum_repository is not a legal parameter in an Ansible task or handler
```

**Comment by ktdreyer** [comment]
> I think next's master tag should be v2.x. What do you think?

That's great, thanks for explaining.

**Comment by leseb** [comment]
@andrewschoen thanks for testing, do you mind having a closer look and submit a fix directly into the branch?
Thanks!

@ktdreyer np! Let's wait a bit until we have a working 1.9 branch then we will merge this one.

**Pull request merged**